xonsh/pyghooks.py

```diff
@@ -1373,54 +1374,80 @@ def color_file(file_path: str, mode: int) -> (Color, str):
     Notes
     -----
 
-    * doesn't handle CA (capability)
-    * doesn't handle LS TARGET mapping
-
+    * implementation follows one authority:
+      https://github.com/coreutils/coreutils/blob/master/src/ls.c#L4879
+    * except:
+    1. does not return 'mi' symlink to missing file unless ln=target,
+      because caller provides an existing file.
+    2. in dircolors, setting type code to '0 or '00' bypasses that test and proceeds to others.
+       In our implementation, setting code to '00' paints the file with no color.
+       This is arguably a bug.
+    3. dircolors with ln=target only colors links to fundamental types, not to *.zoo 
+       (file extension patterns). This function will paint the link with the ultimate file type color.
+       This is arguably better for the user typing a filename on the command line.
     """
 
```

Comment:
One thing with `or` and `mi` is not yet correct I think: in **non**-ln=target mode, symlinks to missing files should be `or` (they seem to be `ln` currently).

I'm not completely sure where is the difference between this code and ls' on this matter. Probably a subtlety in how ln=target is handled by ls, with macros, etc.

Reply:
Thank you once again for providing a great review! I never knew the difference between 'or' and 'mi' and was not able to infer it from what I was seeing in `ls`.